malaria24/ona/models.py

```diff
@@ -54,6 +52,200 @@ def send_digest_email(self):
             html_message=html_content)
 
 
+class NationalDigest(models.Model):
+    """
+    A National Digest of reported cases.
+    """
+    created_at = models.DateTimeField(auto_now_add=True)
+    recipients = models.ManyToManyField('Actor')
+
+    @classmethod
+    def compile_digest(cls):
+        recipients = Actor.objects.filter(
+            role__in=[MANAGER_NATIONAL, MIS],
+            email_address__isnull=False)
+        digest = cls.objects.create()
+        digest.recipients = recipients
+        digest.save()
+        return digest
+
+    def get_digest_email_data(self):
+        provinces = []
+        date = datetime.today()
+        start_week = date - timedelta(date.weekday())
+        end_week = start_week + timedelta(7)
+        week = 'Week ' + str(date.strftime("%U")) + ' ' + str(date.year)
+        # populate provinces
+        for p, p_name in PROVINCES:
+            codes = Facility.objects.filter(province=p).values_list(
+                'facility_code')
+            province_cases = ReportedCase.objects.filter(
+                facility_code__in=codes,
+                created_at__range=[start_week, end_week])
```

Comment:
Can you please add a test for this date range? Just to be sure all cases are covered. i.e are you sure of what day of the week the digest is currently being sent?